flink-runtime/src/main/java/org/apache/flink/runtime/client/DuplicateJobSubmissionException.java

```diff
@@ -25,7 +25,19 @@
  */
 public class DuplicateJobSubmissionException extends JobSubmissionException {
 
-    public DuplicateJobSubmissionException(JobID jobID) {
+    private final boolean terminated;
+
+    public DuplicateJobSubmissionException(JobID jobID, boolean terminated) {
```

Comment:
Personally, I do prefer enums over booleans because they allow to give an expressive name to values. That way it is a bit easier what `DuplicateJobSubmissionException(jobId, JOB_TERMINATED)` means.

Reply:
Agreed with the reasoning, however it also have negatives. Primitive boolean is a simple two state flag. Enums can have multiple states, new ones can be added over time, so it may actually complicate maintenance and by nature they require more sophisticated handling (null checks, unknown states).

If the concern is about "constructor readability", how about replacing this with "named constructors"?

`DuplicateJobSubmissionException.of(jobId)`
`DuplicateJobSubmissionException.ofGloballyTerminated(jobId)`

I have no hard feelings about this one, so if you don't like the proposal, I can stick to enums.

Reply:
Factory methods are a better solution here. They give an explicit name w/o having to fiddle around with booleans.